### PR TITLE
Added a switch that prevents the Makefile from fetching provider creds

### PR DESCRIPTION
**Pull request by mvisonneau**
This can be useful if the user already have access onto the statefile and has configured TF to fetch the creds through Vault instead. eg: https://www.terraform.io/docs/providers/vault/d/aws_access_credentials.html

Implemented it in a fashion that will not break current implementations anyway.

### CODE REVIEW

File: terraform.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ vault_path=${vault_path:-""}
 vault_ttl=${vault_ttl:-"15m"}
 vault_aws_role=${vault_aws_role:-"admin"}
 vault_aws_iam=${vault_aws_iam:-"false"}
+vault_fetch_provider_creds=${vault_fetch_provider_creds:-"true"}
```
Comment (paulRbr) [inline terraform.sh]: what about simply `vault=true`? WDYT?

Comment (mvisonneau) [reply terraform.sh]: Yep, as you like, the thing is that it's only impacting the fetch of credentials at the provider level. It doesn't prevent all Vault actions such as fetching a token with the app roles creds.

Comment (paulRbr) [reply terraform.sh]: True and the `VAULT_ADDR` already acts as a "flag" for activating vault... 🤔 hm

What is the use case of such changes then?

We could mage `vault_path` mandatory to be passed in parameter of the CLI and this variable would act as a flag to fetch provider creds?

Comment (mvisonneau) [reply terraform.sh]: The use case is the following, you run the command from an EC2 instance that already has access onto the S3 bucket, although you want that instance to fetch creds on an alternative aws account that is authorized in Vault which will be done in TF.

**Comment by mvisonneau** [comment]
Actually I figured it would be more explicit to have a flag to specify if we want to get aws creds or not. This would be useful for the stacks that needs access onto the S3 bucket that host the statefile but actually needs to perform action onto a non AWS provider.

**Comment by paulRbr** [comment]
Thanks!

**Pull request merged**